fix(eslint-plugin): [prefer-regexp-exec] respect flags when using `RegExp`

rafaelss95 opened the pull request:
Fixes #3477.

bradzacher reviewed (changes requested):
It looks like there currently aren't any existing tests for `new RegExp` - but I just checked and the rule currently supports its.
It just doesn't do the "global" check on them.

Your change causes the `new RegExp` constructor to be completely ignored - which is not correct.

Instead we should inspect the `new` constructor to extract the second argument to determine if it has a `'g'` in the string

rafaelss95 commented:
> Your change causes the new RegExp constructor to be completely ignored - which is not correct.

@bradzacher Sorry, but I didn't understand what you mean, I've tested it locally and the reports went expected. Note that I had already added a test that covered `new RegExp(...)` for valid cases and I just added another one for invalid ones, just in case.

bradzacher reviewed (approved):
thanks for adding all of those tests!
LGTM!
